From 78cbe3bb1970982adf32c9966b83789301d45cd8 Mon Sep 17 00:00:00 2001
From: Maxence GUILHIN <maxence.guilhin@st.com>
Date: Mon, 10 Jul 2023 15:47:59 +0200
Subject: [PATCH 1/1] tensorflow-lite : fix aarch64 support for XNNPACK

The "-mcpu=" switch might be introduced externaly into CMake: either in <LANG>_FLAGS or
as part of CC, CXX, ASM environmental variables (to be stored in CMAKE_<LANG>_COMPILER_ARG1).
This switch is not compatible with XNNPACK build mechanism and causes the XNNPACK compilation
break due to "unsupported instructions". This switch needs to be removed for XNNPACK
In order to isolate the changes only for XNNPACK and its depencencies, a subfolder is
introduced.

Signed-off-by: Maxence GUILHIN <maxence.guilhin@st.com>
---
 tensorflow/lite/tools/cmake/modules/xnnpack.cmake | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/tensorflow/lite/tools/cmake/modules/xnnpack.cmake b/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
index 24f0674cf92..028fc3a76e2 100644
--- a/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
+++ b/tensorflow/lite/tools/cmake/modules/xnnpack.cmake
@@ -38,6 +38,20 @@ endif()
 set(XNNPACK_BUILD_TESTS OFF CACHE BOOL "Disable XNNPACK test.")
 set(XNNPACK_BUILD_BENCHMARKS OFF CACHE BOOL "Disable XNNPACK benchmarks.")

+#  The "-mcpu=" switch might be introduced externaly into CMake: either in <LANG>_FLAGS or
+#  as part of CC, CXX, ASM environmental variables (to be stored in CMAKE_<LANG>_COMPILER_ARG1).
+#  This switch is not compatible with XNNPACK build mechanism and causes the XNNPACK compilation
+#  break due to "unsupported instructions". This switch needs to be removed for XNNPACK
+#  In order to isolate the changes only for XNNPACK and its depencencies, a subfolder is
+#  introduced.
+
+foreach(FLAG IN ITEMS CMAKE_ASM_FLAGS CMAKE_ASM_COMPILER_ARG1 CMAKE_C_FLAGS CMAKE_C_COMPILER_ARG1 CMAKE_CXX_FLAGS CMAKE_CXX_COMPILER_ARG1)
+  if(${FLAG})
+    string(REGEX REPLACE "-mcpu=[-a-zA-Z0-9_.^$*+?]*" "" _tmp ${${FLAG}})
+    set(${FLAG} ${_tmp})
+  endif()
+endforeach()
+
 # The following line adds project of PTHREADPOOL, FP16 and XNNPACK which are
 # needed to compile XNNPACK delegate of TFLite.
 add_subdirectory(
--
2.25.1
